neuralop/layers/mlp.py

```diff
@@ -44,27 +44,47 @@ def __init__(
             if dropout > 0.0
             else None
         )
-
-        Conv = getattr(nn, f"Conv{n_dim}d")
+        
+        # use nn.Conv1d for everything and roll data along the 1st data dim
         self.fcs = nn.ModuleList()
         for i in range(n_layers):
             if i == 0 and i == (n_layers - 1):
-                self.fcs.append(Conv(self.in_channels, self.out_channels, 1))
+                self.fcs.append(nn.Conv1d(self.in_channels, self.out_channels, 1))
             elif i == 0:
-                self.fcs.append(Conv(self.in_channels, self.hidden_channels, 1))
+                self.fcs.append(nn.Conv1d(self.in_channels, self.hidden_channels, 1))
             elif i == (n_layers - 1):
-                self.fcs.append(Conv(self.hidden_channels, self.out_channels, 1))
+                self.fcs.append(nn.Conv1d(self.hidden_channels, self.out_channels, 1))
             else:
-                self.fcs.append(Conv(self.hidden_channels, self.hidden_channels, 1))
+                self.fcs.append(nn.Conv1d(self.hidden_channels, self.hidden_channels, 1))
 
     def forward(self, x):
+        # if x's data is >1d, reshape into 1d b, c, x
+        reshaped = False
+        size = list(x.shape)
+        if x.ndim > 3:  # batch, channels, x1, x2... extra dims
+            # flatten last data dims
+            if x.is_contiguous():
+                # view only works on contiguous tensors but is better
+                x = x.view(*size[:2], -1) 
+            else:
+                x = x.reshape((*size[:2], -1)) 
+            reshaped = True
+
         for i, fc in enumerate(self.fcs):
             x = fc(x)
             if i < self.n_layers - 1:
                 x = self.non_linearity(x)
             if self.dropout is not None:
                 x = self.dropout[i](x)
 
+        # if x was an Nd tensor reshaped into 3d, undo the reshaping
+        if reshaped:
+            # same logic as above: only contiguous tensors can call .view()
+            if x.is_contiguous:
+                x = x.view(size[0], self.out_channels, *size[2:])
+            else:
+                x = x.reshape((size[0], self.out_channels, *size[2:]))
```

Comment:
I think here it may be clearer to just use reshape all the time: if it wasn't reshaped, nothing happens. If a view is possible it will be used (most likely, see above, unless the input isn't contiguous), otherwise it will be reshaped. 

We should comment on the logic behind these decisions though!

Reply:
Gotcha! Moved to just `reshape` and commented about it